### ISSUE TITLE
Node.js Support

### ISSUE THREAD

**Issue by gtrak** [clj-commons/hickory#17]
Not sure if this is a good idea... 

I'm trying to get a nodejs test-runner working for a node-webkit project, however hickory relies on DOM APIs not present in node.js (works fine in the real app).

Alternatively, I can try to find a way to use node-webkit as a test runner.

Actual Error:

```
  return Node[[cljs.core.str(a), cljs.core.str("_NODE")].join("")];
         ^
ReferenceError: Node is not defined
    at hickory.core.node_type (/home/gary/my-project/target/testing.js:31175:10)
    at Object.<anonymous> (/home/gary/my-project/target/testing.js:31177:49)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:902:3
```

**Comment by hzhu** [comment]
Bump. @davidsantiago @jeluard 

Hey guys, I'm working on a server side rendered app using Reagent, CLJS, and Node. https://github.com/savaki/reagent-nodejs

Node was blowing up and I realized that Hickory relies on the DOM, which isn't available in Node. I'm getting the exact exception as @gtrak 

Anyone been able to solve this issue? I see that @skrat made a pull request but hasn't been merged.

**Comment by jeluard** [comment]
I understand @skrat is using [libxml-dom](https://github.com/skrat/libxml-dom) as a polyfill to the browser DOM API. I might be wrong so take that with a grain of salt.
This sounds like the best approach and doesn't require changes to hickory.

**Comment by hzhu** [comment]
Thanks, libxml-dom seems to be helping: loading hickory in Node actually works now.

But once I start using `parse-fragment` an exception `TypeError: Cannot call method 'createHTMLDocument' of undefined`

    hickory.core.parse_dom_with_write = (function parse_dom_with_write(s) {
        var doc = document.implementation.createHTMLDocument("");
        var doctype_el = (doc["doctype"]);
        if(cljs.core.truth_(hickory.core.extract_doctype.call(null,s))){
    } else {
        hickory.core.remove_el.call(null,doctype_el);
    }

Any suggestions @skrat ?

**Comment by skrat** [comment]
@hzhu Because there's no `implementation` property on `document` object. Feel free to fork `libxml-dom` and send in a PR. I might do this myself but can't give you any estimate.

**Comment by olivergg** [comment]
Just for the record, hickory does not work either in a React Native environment (using re-natal on an android device)... I've got the same error as above (`Node is not defined`).

**Comment by TerjeNorderhaug** [comment]
Fixed in pull request #33 which allows hickory to load on nodejs.

**Comment by slipset** [comment]
Closing this as #33 is merged
